### PR TITLE
added RESTful api with filterable variation api

### PR DESCRIPTION
**Pull request by Kaibu**
default RESTful api's for size, sizegroup and products, filterable for variants

dect plan: select product -> get available sizes for product and left amounts

configured with anonymous read for now

### CODE REVIEW

File: hagrid/api/urls.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+router = routers.DefaultRouter()
+router.register(r'product', ProductViewSet)
+router.register(r'size', SizeViewSet)
+router.register(r'sizegroup', SizeGroupViewSet)
+
+urlpatterns = [
+    path('v1/auth/', include('rest_framework.urls')),
+    path('v1/', include(router.urls)),
+    path('v1/variants', VariationProductDetail.as_view(), name='api_variants'),
```
Comment (felixrindt) [inline hagrid/api/urls.py]: Should be "variation" and not "variant" two times on this line.

File: hagrid/api/serializers.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+class SizeAPISerializer(serializers.ModelSerializer):
+    group = SizeGroupAPISerializer()
+    class Meta:
+        model = Size
+        fields = '__all__'
+
+
+class VariationAPISerializer(serializers.ModelSerializer):
+    size = SizeAPISerializer()
```
Comment (felixrindt) [inline hagrid/api/serializers.py]: Do you think we should also include `product = ProductAPISerializer()`, so a single call to this endpoint is enough for most applications?

File: hagrid/settings.py [hunk 3]
```diff
@@ -118,6 +120,12 @@
     },
 ]
 
+REST_FRAMEWORK = {
+    'DEFAULT_PERMISSION_CLASSES': [
+        'rest_framework.permissions.DjangoModelPermissionsOrAnonReadOnly'
```
Comment (felixrindt) [inline hagrid/settings.py]: As we do not need API write permissions, even for logged in users, I'd prefer to explicitly only allow reads from *anyone*.

File: hagrid/api/urls.py (hunk 1 above)
Comment (Kaibu) [reply hagrid/api/urls.py]: fixed

File: hagrid/settings.py (hunk 3 above)
Comment (Kaibu) [reply hagrid/settings.py]: Added custom permission class to only allow same_methods (GET, OPTIONS, HEAD)

File: hagrid/api/serializers.py (hunk 2 above)
Comment (Kaibu) [reply hagrid/api/serializers.py]: Sure, added

**Comment by felixrindt** [comment]
Thanks! :tada:

**Pull request merged**